src/instructlab/cli/data/list.py

```diff
@@ -39,5 +67,38 @@ def list_datasets(dataset_dirs):
         click.secho(f"Failed to list datasets with exception: {exc}")
         raise click.exceptions.Exit(1)
 
+    grouped_data = defaultdict(list)
+    timestamp_to_model = {}
+    general_files = []
+
+    # Process model name files e.g. start with test_, train_, messages_
+    for item in data:
+        filename = item[0]
+        group_key, is_general, timestamp = extract_model_and_timestamp(filename)
+
+        if group_key:
+            if is_general:
+                general_files.append((item, timestamp))
+            else:
+                model_name = group_key.split(" ", 1)[0]
+                expected_key = f"{model_name} {timestamp}"
```

Comment:
I am still curious, is expected_key ever different from group_key in this case?  If it isn't, why not just use group_key?